src/System.Management.Automation/namespaces/FileSystemProvider.cs

```diff
@@ -6074,24 +6074,43 @@ protected override bool IsItemContainer(string path)
         /// <param name="force">If true, force move the directory, overwriting anything at the destination.</param>
         private void MoveDirectoryInfoUnchecked(DirectoryInfo directory, string destinationPath, bool force)
         {
-            try
+            if (!IsSameWindowsVolume(directory.FullName, destinationPath))
             {
-                if (InternalTestHooks.ThrowExdevErrorOnMoveDirectory)
-                {
-                    throw new IOException("Invalid cross-device link", hresult: MOVE_FAILED_ERROR);
-                }
-
-                directory.MoveTo(destinationPath);
+                CopyAndDelete(directory, destinationPath, force);
```

Comment:
You could just early return here, and remove the `else` block.

Reply:
With the simplification of the code, this is no longer relevant as there isn't an `else` clause